### PR TITLE
Add restore retry

### LINKED ISSUE #417: Consider adding retry behavior in transient network failures
In a recent failed test run I see this:
```console
[2019/03/20 16:10:53][INFO] $ pushd "/home/ddit/dotnetbuild/work/cbdb4e6b-e558-4715-ac11-769e2a0162ab/Work/51a9e9bd-019d-491a-b4a8-9990a61b7b8e/Exec/src/benchmarks/micro"
[2019/03/20 16:10:53][INFO] $ dotnet restore /home/ddit/dotnetbuild/work/cbdb4e6b-e558-4715-ac11-769e2a0162ab/Work/51a9e9bd-019d-491a-b4a8-9990a61b7b8e/Exec/src/benchmarks/micro/MicroBenchmarks.csproj --packages /home/ddit/dotnetbuild/work/cbdb4e6b-e558-4715-ac11-769e2a0162ab/Work/51a9e9bd-019d-491a-b4a8-9990a61b7b8e/Exec/artifacts/packages
[2019/03/20 16:10:54][INFO]   Persisting no-op dg to /home/ddit/dotnetbuild/work/cbdb4e6b-e558-4715-ac11-769e2a0162ab/Work/51a9e9bd-019d-491a-b4a8-9990a61b7b8e/Exec/artifacts/obj/MicroBenchmarks/MicroBenchmarks.csproj.nuget.dgspec.json
[2019/03/20 16:10:54][INFO]   Persisting no-op dg to /home/ddit/dotnetbuild/work/cbdb4e6b-e558-4715-ac11-769e2a0162ab/Work/51a9e9bd-019d-491a-b4a8-9990a61b7b8e/Exec/artifacts/obj/BenchmarkDotNet.Extensions/BenchmarkDotNet.Extensions.csproj.nuget.dgspec.json
[2019/03/20 16:19:12][INFO] /home/ddit/dotnetbuild/work/cbdb4e6b-e558-4715-ac11-769e2a0162ab/Work/51a9e9bd-019d-491a-b4a8-9990a61b7b8e/Exec/tools/dotnet/x64/sdk/3.0.100-preview4-010918/NuGet.targets(119,5): error : Unable to load the service index for source https://dotnet.myget.org/F/symreader-converter/api/v3/index.json. [/home/ddit/dotnetbuild/work/cbdb4e6b-e558-4715-ac11-769e2a0162ab/Work/51a9e9bd-019d-491a-b4a8-9990a61b7b8e/Exec/src/benchmarks/micro/MicroBenchmarks.csproj]
[2019/03/20 16:19:12][INFO] /home/ddit/dotnetbuild/work/cbdb4e6b-e558-4715-ac11-769e2a0162ab/Work/51a9e9bd-019d-491a-b4a8-9990a61b7b8e/Exec/tools/dotnet/x64/sdk/3.0.100-preview4-010918/NuGet.targets(119,5): error :   An error occurred while sending the request. [/home/ddit/dotnetbuild/work/cbdb4e6b-e558-4715-ac11-769e2a0162ab/Work/51a9e9bd-019d-491a-b4a8-9990a61b7b8e/Exec/src/benchmarks/micro/MicroBenchmarks.csproj]
[2019/03/20 16:19:12][INFO] /home/ddit/dotnetbuild/work/cbdb4e6b-e558-4715-ac11-769e2a0162ab/Work/51a9e9bd-019d-491a-b4a8-9990a61b7b8e/Exec/tools/dotnet/x64/sdk/3.0.100-preview4-010918/NuGet.targets(119,5): error :   Unable to read data from the transport connection: Connection reset by peer. [/home/ddit/dotnetbuild/work/cbdb4e6b-e558-4715-ac11-769e2a0162ab/Work/51a9e9bd-019d-491a-b4a8-9990a61b7b8e/Exec/src/benchmarks/micro/MicroBenchmarks.csproj]
[2019/03/20 16:19:12][INFO] /home/ddit/dotnetbuild/work/cbdb4e6b-e558-4715-ac11-769e2a0162ab/Work/51a9e9bd-019d-491a-b4a8-9990a61b7b8e/Exec/tools/dotnet/x64/sdk/3.0.100-preview4-010918/NuGet.targets(119,5): error :   Connection reset by peer [/home/ddit/dotnetbuild/work/cbdb4e6b-e558-4715-ac11-769e2a0162ab/Work/51a9e9bd-019d-491a-b4a8-9990a61b7b8e/Exec/src/benchmarks/micro/MicroBenchmarks.csproj]
[2019/03/20 16:19:12][ERROR] Process exited with status 1
```

That looks like what is probably a transient failure. It is probably a good idea to go through our automation for places like this and add a small retry so we don't lose test runs for simple network trouble.

### PR DESCRIPTION
**Pull request by billwert**
Fixes #417

### CODE REVIEW

File: scripts/performance/common.py [hunk 1]
```diff
@@ -180,9 +181,20 @@ def run(self, working_directory: str = None) -> None:
...
-                        raise CalledProcessError(
-                            proc.returncode, quoted_cmdline)
+        retrycount = 0
+        (returncode, quoted_cmdline) = self.__runinternal(working_directory)
+        while returncode not in self.success_exit_codes and retrycount < self.__retry:
+            (returncode, _) = self.__runinternal(working_directory)
+            retrycount += 1
+
+        if returncode not in self.success_exit_codes:
+            getLogger().error(
```
Comment (ooooolivia) [inline scripts/performance/common.py:197]: nit: remove the line break?

File: scripts/performance/common.py [hunk 2]
```diff
@@ -180,9 +181,20 @@ def run(self, working_directory: str = None) -> None:
...
+                "Process exited with status %s", returncode)
+            raise CalledProcessError(
```
Comment (ooooolivia) [inline scripts/performance/common.py:199]: nit: remove the line break?

Comment (ooooolivia) [reply scripts/performance/common.py:199]: Actually there are many line breaks in this file look weird to me and have been there before your change. Not sure if that's intentional but feel free to keep/change all of them.

Comment (billwert) [reply scripts/performance/common.py:199]: Yeah I didn't actually add these line breaks. It just looks that way because of how the diff winds up.

**Pull request merged**